app/lib/actions/katello/content_view_environment/destroy.rb

```diff
@@ -7,6 +7,12 @@ def plan(cv_env, options = {})
           organization_destroy = options.fetch(:organization_destroy, false)
           content_view = cv_env.content_view
           environment = cv_env.environment
+          if cv_env.activation_keys.any?(&:multi_content_view_environment?)
+            remove_activation_key_associations(cv_env)
+            remove_content_view_environment_association(content_view, environment)
+            plan_self(:id => cv_env.id, :docker_cleanup => false)
+            return true
+          end
```

Comment:
I'm questioning whether this logic is okay here within the main `ContentViewEnvironment::Destroy` action, or if it would be better within a new sub-action.. @sjha4 thoughts?

Reply:
We can refactor it more so the logic separation is clearer to read.